fea-rs/src/compile/variations.rs

```diff
@@ -14,9 +19,20 @@ pub trait VariationInfo {
     fn axis_info(&self, axis_tag: Tag) -> Option<AxisInfo>;
 
     /// Return the normalized value for a user coordinate for the given axis.
+    ///
+    /// NOTE: This is only used for computing the normalized values for ConditionSets.
     fn normalize_coordinate(&self, axis_tag: Tag, value: Fixed) -> F2Dot14;
+
+    /// return the default value, and then the regions and their deltas.
+    fn resolve_variable_metric(
+        &self,
+        locations: &HashMap<UserLocation, i16>,
+    ) -> (i16, Vec<(VariationRegion, i16)>);
```

Comment:
Result please

Reply:
Wasn't really sure what the error conditions are. Easy enough to make the error an associated type and let the implementor fill it in, though.

Reply:
Here too it would be nice if we could share types; var model has it's own region type https://github.com/googlefonts/fontc/blob/735033d3d23900424cfc84e241277c7d6bd853fa/fontir/src/variations.rs#L399.

Reply:
The return type is throwing me, what is the i16? Shouldn't it be the same as  VariationModel::deltas, so the success result is `Vec<(VariationRegion, Vec<V>)>` and V is i16 based on what you have here?

Reply:
For reference https://github.com/googlefonts/fontc/pull/345/files#diff-3a9c81a9a13ca7222965ea3f55a21d3bb43315a54e222f866baccf9c974bda79 shows my first pass at implementing this method

Reply:
It's very possible I got this wrong, I'm still wrapping my head around the delta model. I think that the API of VariationModel::deltas is assuming that it is computing deltas for multiple points at a time (as in a glyph) but for variable metrics I'm only computing a single value at a time.


> Here too it would be nice if we could share types; var model has it's own region type https://github.com/googlefonts/fontc/blob/735033d3d23900424cfc84e241277c7d6bd853fa/fontir/src/variations.rs#L399.

I did look at that, but ended up not wanting to bring in fontir as a dep, and I do ultiamtely need this stuff to end up in the `write-fonts` formats, so I just used those.

Reply:
I agree fea-rs should not depend on fontir. I would argue it could depend on fontdrasil and we could move some types there.

Reply:
Concretely, could we make the return `-> Result<Vec<(VariationRegion, i16)>, ErrorType>` instead of `-> (i16, Vec<(VariationRegion, i16)>)`?

Reply:
And there's some way for me to determine the default value given that result?

Reply:
Err no. No there is not, I'm apparently too used to glyfs where the default is 0. Forget that idea!